src/appleseed/foundation/array/algorithm.cpp

```diff
@@ -47,7 +47,9 @@ namespace
         ArrayView<SrcType> src(array);
         ArrayRef<DstType> dst(tmp);
 
-        std::copy(src.begin(), src.end(), std::back_inserter(dst));
+        for (const auto value : src)
```

Comment:
Make a const ref here in case the value type is not a single scalar?

Reply:
It is always going to be an integer type.